blockchain/blockchain_test.go

```diff
@@ -797,3 +797,47 @@ func TestMakeStateDiffForEmptyBlock(t *testing.T) {
 		assert.Equal(t, blockHash, sd.StorageDiffs[*storageContractAddr][felt.Zero])
 	})
 }
+
+func TestFinalize(t *testing.T) {
+	testDB := pebble.NewMemTest(t)
+	chain := blockchain.New(testDB, utils.Mainnet, utils.NewNopZapLogger())
+
+	receipts := make([]*core.TransactionReceipt, 0)
+	pendingBlock := &core.Block{
+		Header: &core.Header{
+			ParentHash:       &felt.Zero,
+			SequencerAddress: utils.HexToFelt(t, "0x1337"),
+			Timestamp:        0,
+			ProtocolVersion:  "0.0.0",
+			EventsBloom:      core.EventsBloom(receipts),
+			GasPrice:         utils.HexToFelt(t, "0x1"),
+			GasPriceSTRK:     utils.HexToFelt(t, "0x2"),
+		},
+		Transactions: make([]core.Transaction, 0),
+		Receipts:     receipts,
+	}
+
+	stateDiff, err := blockchain.MakeStateDiffForEmptyBlock(chain, 0)
+	require.NoError(t, err)
+
+	pendingGenesis := &blockchain.Pending{
+		Block: pendingBlock,
+		StateUpdate: &core.StateUpdate{
+			OldRoot:   &felt.Zero,
+			StateDiff: stateDiff,
+		},
+		NewClasses: make(map[felt.Felt]core.Class, 0),
+	}
+
+	require.NoError(t, chain.Finalize(pendingGenesis))
+	h, err := chain.Height()
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), h)
+
+	pending, err := chain.Pending()
+	require.NoError(t, err)
+	require.NoError(t, chain.Finalize(&pending))
+	h, err = chain.Height()
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), h)
+}
```

Comment:
Can we test that the header, txs, and receipts have been stored correctly as well?

Reply:
Done, that actually uncovered a bug when unmarshaling empty blocks from storage:D